Option resolveJsonModule doesn't work without esModuleInterop

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨

Half of all issues filed here are duplicates, answered in the FAQ, or not appropriate for the bug tracker. Even if you think you've found a *bug*, please read the FAQ first, especially the Common "Bugs" That Aren't Bugs section!

Please help us by doing the following steps before logging an issue:
  * Search: https://github.com/Microsoft/TypeScript/search?type=Issues
  * Read the FAQ: https://github.com/Microsoft/TypeScript/wiki/FAQ

Please fill in the *entire* template below.
-->

<!-- Please try to reproduce the issue with `typescript@next`. It may have already been fixed. -->
**TypeScript Version:**  2.9.2, 3.0.0-dev.20180703

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms:** 

resolveJsonModule, esModuleInterop

**Code**

```ts
import * as test from './test.json';
```

tsconfig.json:

```json
{
    "compilerOptions": {
        "module": "commonjs",
        "moduleResolution": "node",
        "target": "es2017",
        "lib": ["es2017"],
        "strict": true,
        "sourceMap": true,
        "noEmitOnError": true,
        "baseUrl": ".",
        "resolveJsonModule": true,
        "outDir": "out"
    }
}
```

**Expected behavior:**

No errors.

**Actual behavior:**

Compile error:

```
$ tsc
test.ts:1:23 - error TS2497: Module '"/home/kostya/tmp/resolve-json-test/test"' resolves to a non-module entity and cannot be imported using this construct.

1 import * as test from './test.json';
                        ~~~~~~~~~~~~~
```

***

When I add `esModuleInterop` option to `tsconfig.json`:

```diff
diff --git a/tsconfig.json b/tsconfig.json
index 7f1afb8..e949135 100644
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -9,6 +9,7 @@
         "noEmitOnError": true,
         "baseUrl": ".",
         "resolveJsonModule": true,
+        "esModuleInterop": true,
         "outDir": "out"
     }
 }
```

and change import statement to:

```diff
diff --git a/test.ts b/test.ts
index 07bb9b7..dddcffb 100644
--- a/test.ts
+++ b/test.ts
@@ -1 +1 @@
-import * as test from './test.json';
+import test from './test.json';
```

it works. But with  `esModuleInterop` option I need to change many imports in my project, which is undesirable.

**Playground Link:** <!-- A link to a TypeScript Playground "Share" link which demonstrates this behavior --> 

**Related Issues:** <!-- Did you find other bugs that looked similar? -->

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -2268,7 +2268,7 @@ namespace ts {
         function resolveESModuleSymbol(moduleSymbol: Symbol | undefined, referencingLocation: Node, dontResolveAlias: boolean): Symbol | undefined {
             const symbol = resolveExternalModuleSymbol(moduleSymbol, dontResolveAlias);
             if (!dontResolveAlias && symbol) {
-                if (!(symbol.flags & (SymbolFlags.Module | SymbolFlags.Variable))) {
+                if (!(symbol.flags & (SymbolFlags.Module | SymbolFlags.Variable)) && !getDeclarationOfKind(symbol, SyntaxKind.SourceFile)) {
                     error(referencingLocation, Diagnostics.Module_0_resolves_to_a_non_module_entity_and_cannot_be_imported_using_this_construct, symbolToString(moduleSymbol!));
                     return symbol;
                 }

--- a/tests/baselines/reference/requireOfJsonFileWithoutEsModuleInterop.js
+++ b/tests/baselines/reference/requireOfJsonFileWithoutEsModuleInterop.js
@@ -1,0 +1,19 @@
+//// [tests/cases/compiler/requireOfJsonFileWithoutEsModuleInterop.ts] ////
+
+//// [file1.ts]
+import * as test from "./test.json"
+
+//// [test.json]
+{
+    "a": true,
+    "b": "hello"
+}
+
+//// [out/test.json]
+{
+    "a": true,
+    "b": "hello"
+}
+//// [out/file1.js]
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });

--- a/tests/baselines/reference/requireOfJsonFileWithoutEsModuleInterop.symbols
+++ b/tests/baselines/reference/requireOfJsonFileWithoutEsModuleInterop.symbols
@@ -1,0 +1,12 @@
+=== tests/cases/compiler/file1.ts ===
+import * as test from "./test.json"
+>test : Symbol(test, Decl(file1.ts, 0, 6))
+
+=== tests/cases/compiler/test.json ===
+{
+    "a": true,
+>"a" : Symbol("a", Decl(test.json, 0, 1))
+
+    "b": "hello"
+>"b" : Symbol("b", Decl(test.json, 1, 14))
+}

--- a/tests/baselines/reference/requireOfJsonFileWithoutEsModuleInterop.types
+++ b/tests/baselines/reference/requireOfJsonFileWithoutEsModuleInterop.types
@@ -1,0 +1,16 @@
+=== tests/cases/compiler/file1.ts ===
+import * as test from "./test.json"
+>test : { "a": boolean; "b": string; }
+
+=== tests/cases/compiler/test.json ===
+{
+>{    "a": true,    "b": "hello"} : { "a": boolean; "b": string; }
+
+    "a": true,
+>"a" : boolean
+>true : true
+
+    "b": "hello"
+>"b" : string
+>"hello" : "hello"
+}

--- a/tests/cases/compiler/requireOfJsonFileWithoutEsModuleInterop.ts
+++ b/tests/cases/compiler/requireOfJsonFileWithoutEsModuleInterop.ts
@@ -1,0 +1,16 @@
+// @module: commonjs
+// @moduleResolution: node
+// @target:es2017
+// @strict: true
+// @resolveJsonModule: true
+// @outdir: out/
+// @fullEmitPaths: true
+
+// @Filename: file1.ts
+import * as test from "./test.json"
+
+// @Filename: test.json
+{
+    "a": true,
+    "b": "hello"
+}

